packages/medusa/src/loaders/helpers/routing/__tests__/index.spec.ts

```diff
@@ -155,34 +155,6 @@ describe("RoutesLoader", function () {
       expect(res.text).toBe("GET /admin/unprotected")
     })
 
-    it("should return 200 when customer is authenticated", async () => {
```

Comment:
@olivermrbl @adrien2p after looking at the code, I think it makes sense to not apply any authentication middleware for the storefront.

Currently we have 3 routes that require authentication
- `POST /store/customers` with allowUnregistered
- `ALL /store/customers/me*`
- `GET /store/orders`

It is very weird if we apply the middleware globally for some routes, but not others, so I think it's easier to reason and understand if we say:
**We apply auth middleware globally for admin**
**We don't apply auth middleware globally for store**

The auth context will be filled in on all routes btw, this is just for restricting access.
WDYT?

Reply:
Yes I think it makes sense and looks closer to what we had no? we still have restricted the few store end points that requires it but then the user can restrict what he needs if necessary

Reply:
No strong opinions on this, but I do think restricting `/store/customers/me` is a good way of maintaining consistency between the developers' customizations and our API routes. It's also an approach that makes sense (so not difficult for developers to understand/expect). I do agree it doesn't need to be three different route paths/patterns though.

Reply:
Yes, this sounds right to me. It doesn't make sense to have the store-facing API authenticated by default.

Reply:
Maybe I wasn't clear, but what I meant is, it is better to not apply global auth middleware for `store` at all (before we did have it applied like that). The effect is the same, we would still require auth for `store/customers/me`, but only because it's applied in `customers/middleware.ts`.

Reply:
Ah ok I misunderstood then 😄 sounds good to me 👍🏻